net/dns/resolver/tsdns_test.go

```diff
@@ -373,6 +384,14 @@ func TestResolveLocal(t *testing.T) {
 			if ip != tt.ip {
 				t.Errorf("ip = %v; want %v", ip, tt.ip)
 			}
+
+			allocs := testing.AllocsPerRun(1000, func() {
+				r.resolveLocal(tt.qname, tt.qtype)
+			})
+			// 4via6 paths have 1 alloc, the rest should be 0
+			if allocs > 1 {
```

Comment:
add an alloc test without 4via6 (with just wildcard matching) to see that it's zero